hoot-core/src/main/cpp/hoot/core/conflate/DuplicateWayRemover.cpp

```diff
@@ -163,14 +179,34 @@ bool DuplicateWayRemover::_isCandidateWay(const ConstWayPtr& w) const
   return result;
 }
 
-void DuplicateWayRemover::_removeDuplicateNodes(shared_ptr<Way> w1, shared_ptr<Way> w2)
+void DuplicateWayRemover::_updateWayNameTags(shared_ptr<Way> way1, shared_ptr<Way> way2)
 {
+  //merge name tags for these two ways, if they aren't already identical
+  double namesScore = 0.0;
+  double weight = 0.0;
+  TagComparator::getInstance().compareNames(way1->getTags(), way2->getTags(), namesScore, weight);
```

Comment:
Yes, you can get a 1.0 from compare names even if all the names aren't identical. I think you're right, this should use a full tag merge.

See my comments by line 143/144.